hicommonwealth/commonwealth
# Dissolved `<SessionKeyRevalidationModal/>` into `<AuthModal/>` + updated auth options for session key revalidation modal variant

## Link to Issue
Closes: https://github.com/hicommonwealth/commonwealth/issues/8029

## Description of Changes
- Dissolved `<SessionKeyRevalidationModal/>` into `<AuthModal/>`
- Updated auth option for session key revalidation modal variant of auth modal, now only the auth option that matches the auth method of the expired session will be displayed
- Removed stale components -> `useSessionRevalidationModal()`, `<SessionRevalidationModal/>`, `<CWAuthButton/>`, `<CWNoAuthMethodsAvailable/>` and `<CWWalletsList/>`

## "How We Fixed It"
N/A

## Test Plan
- Enable `ENFORCE_SESSION_KEYS`
- Login
- Create a thread
- Add a comment
- Open the browser console and clear local storage ex: `localStorage.clear()` -- this is for manually expiring session keys in frontend
- Now reload the page and try to create another comment or thread
- Verify that the session revalidation modal is shown with the correct auth option (the auth option you used to login earlier)

<img width="570" alt="image" src="https://github.com/hicommonwealth/commonwealth/assets/51641047/3d2ea495-e8d2-41ee-9f59-e73572afe5bb">


## Deployment Plan
N/A

## Other Considerations
- Not all auth-requiring actions have session keys logic, this PR doesn't attempt to add/fix all of those. The focus of this PR is just to dissolve the existing logic of `<SessionKeyRevalidationModal/>` into `<AuthModal/>`
- ~~ATM, this PR doesn't handle the address mismatch cases when revalidating sessions, this needs to be updated when #7395 is merged~~ -- this is now updated after the canvas migration PR was merged
